clap_man/src/render.rs

```diff
@@ -0,0 +1,195 @@
+use std::io::Write;
+
+use crate::man::{bold, document_title, escape, indent, italic, list, section_title};
+
+use clap::ArgSettings;
+
+fn app_name(app: &clap::App) -> String {
+    if let Some(name) = app.get_bin_name() {
+        escape(name)
+    } else {
+        escape(app.get_name())
+    }
+}
+
+fn subcommand_heading(app: &clap::App) -> String {
+    match app.get_subommand_help_heading() {
+        Some(title) => escape(title),
+        None => "Subcommands".to_string(),
+    }
+}
+
+pub(crate) fn header(app: &clap::App, section: i8, manual: Option<String>, buf: &mut dyn Write) {
+    write!(buf, "{}", document_title(app, section, manual, None)).unwrap();
+}
+
+pub(crate) fn about(app: &clap::App, buf: &mut dyn Write) {
+    write!(buf, "{}", section_title("Name")).unwrap();
+    write!(buf, "{}", bold(&app_name(app))).unwrap();
+
+    if let Some(about) = app.get_about() {
+        write!(buf, r" \- {}", escape(about)).unwrap();
+    }
+
+    writeln!(buf).unwrap();
+}
+
+pub(crate) fn description(app: &clap::App, buf: &mut dyn Write) {
+    if let Some(about) = app.get_long_about() {
+        writeln!(buf, "{}{}", section_title("Description"), escape(about)).unwrap();
+    }
+}
+
+pub(crate) fn synopsis(app: &clap::App, buf: &mut dyn Write) {
+    write!(
+        buf,
+        "{}{} ",
+        section_title("Synopsis"),
+        italic(&app_name(app))
+    )
+    .unwrap();
+
+    for opt in app.get_arguments() {
+        let options = match (opt.get_short(), opt.get_long()) {
+            (Some(short), Some(long)) => format!("[-{}|--{}]", short, long),
+            (Some(short), None) => format!("[-{}]", short),
+            (None, Some(long)) => format!("[--{}]", long),
+            (None, None) => "".to_string(),
```

Comment:
Shouldn't we only use `[]` if its optional?

Reply:
Also, this should be showing value names

Reply:
I've updated this for arguments, and they show the value names (though not fully correctly yet for ones with multible ones), same as below.